tensorflow/compiler/mlir/tools/kernel_gen/transforms/tf_to_jit_invocations.cc

```diff
@@ -115,20 +119,33 @@ struct TFToI64JITInvocationForLargeTensorsPattern : public RewritePattern {
 
   LogicalResult matchAndRewrite(Operation *op,
                                 PatternRewriter &rewriter) const override {
-    if (!IsUnaryTFOperation(op) ||
+    if ((!IsUnaryTFOperation(op) && !IsBinaryTFOperation(op)) ||
         !llvm::isa<func::FuncOp>(op->getParentOp())) {
       return failure();
     }
 
     // Create large argument condition.
     auto loc = op->getLoc();
-    auto arg = op->getOperands().front();
-    auto shape = rewriter.create<shape::ShapeOfOp>(loc, arg);
-    auto num_elems = rewriter.create<shape::NumElementsOp>(loc, shape);
+    auto arg_1 = op->getOperands().front();
+    auto shape_1 = rewriter.create<shape::ShapeOfOp>(loc, arg_1);
+    auto num_elems_1 = rewriter.create<shape::NumElementsOp>(loc, shape_1);
     Value cst_i32_limit =
         rewriter.create<arith::ConstantIndexOp>(loc, i32Limit);
-    Value large_tensor_predicate = rewriter.create<arith::CmpIOp>(
-        loc, arith::CmpIPredicate::sgt, num_elems, cst_i32_limit);
+    Value large_tensor_predicate, large_tensor_predicate_first_arg;
```

Comment:
To avoid the uninitalized `Value`s you can just create the `large_tensor_predicate` and in case of a binary op update that value.

Reply:
done.